File: Longest-substring-Without-Repeating.cpp
#include <iostream>
#include "bits/stdc++.h"
using namespace std;

class Solution{
public:
	int getLong(string text){
		unordered_set<char> s;
		for (auto i : text) {
			if(s.find(i) != s.end()){
				return s.size();
			}
			else{
				s.insert(i);
			}
		}
		return s.size();
	}
	int lengthOfLongestSubstring(string text){
		int longest = 0;
		for (int i = 0; i < text.length(); ++i) {
			longest = max(longest, getLong(text.substr(i)));
		}
		return longest;
	}
};

File: Longest-Substring-Without-Repeating.py
class Solution:
    def findLong(self,text):
        s = set()
        long = 0
        for x in text:
            if x in s:
                return len(s)
            else:
                s.add(x)
        return len(s)
    def lengthOfLongestSubstring(self, text: str) -> int:
        i = 0
        longest = 0
        while i<len(text)-longest:
            longest = max(longest,self.findLong(text[i:]))
            i += 1
        return longest